launch/launch/exit_handler.py

```diff
@@ -23,17 +23,34 @@ def __init__(self, launch_state, task_state):
         self.task_state = task_state
 
 
-def default_exit_handler(context, ignore_returncode=False):
+def default_exit_handler(
+        context,
+        ignore_returncode=False,
+        teardown_on_error_only=False,
+        trigger_teardown=True):
     """
     Trigger teardown of launch.
 
     Use the returncode of the task for the launch if the launch was not already tearing down.
 
     @param ignore_returncode: If True, then ignore process return code.
+    @param teardown_on_error_only: If True, trigger launch context teardown
+    on exit if the return code indicates an error. This is mutually not
+    relevant if ignore_returncode is True or trigger_teardown is False.
+    @param trigger_teardown: If True, trigger launch context teardown
+    on exit if not already triggered.
```

Comment:
The entanglement of the different keyword arguments seems pretty complicated to me. At least to me even after reading the doc it is not straight forward what to pass. Maybe the arguments can be changed somehow or it would be easier to have a separate exit handler if the behavior is so different from the default.